lib/galaxy/datatypes/text.py

```diff
@@ -280,9 +280,10 @@ def _transform_dict_list_ids(dict_list):
                         if b_name == "columns" and metadata_value:
                             keep_columns = set()
                             for column in metadata_value:
-                                for k, v in column['metadata'].items():
-                                    if v is not None:
-                                        keep_columns.add(k)
+                                if column['metadata'] is not None:
+                                    for k, v in column['metadata'].items():
```

Comment:
you could also do `column.get('metadata', {}).items():`. I'm not sure, @nsoranzo you have opinions on these things :)

Reply:
Both are probably fine, @bebatut's version is a bit more strict, since it requires `'metadata'` to be a key in the dict.